packages/core-state/src/wallets/wallet-manager.ts

```diff
@@ -242,114 +242,53 @@ export class WalletManager implements State.IWalletManager {
     public buildVoteBalances(): void {
         for (const voter of this.allByPublicKey()) {
             if (voter.hasVoted()) {
-                const delegate: State.IWallet = this.findByPublicKey(voter.getAttribute<string>("vote"));
-                const voteBalance: Utils.BigNumber = delegate.getAttribute("delegate.voteBalance");
-                const lockedBalance = voter.getAttribute("htlc.lockedBalance", Utils.BigNumber.ZERO);
-                delegate.setAttribute("delegate.voteBalance", voteBalance.plus(voter.balance).plus(lockedBalance));
+                this.increaseDelegateVoteBalance(voter, voter.balance);
+                this.increaseDelegateVoteBalance(voter, voter.getAttribute("htlc.lockedBalance", Utils.BigNumber.ZERO));
             }
         }
     }
 
     public async applyBlock(block: Interfaces.IBlock): Promise<void> {
         this.currentBlock = block;
-        const generatorPublicKey: string = block.data.generatorPublicKey;
-
-        let delegate: State.IWallet;
-        if (!this.has(generatorPublicKey)) {
-            const generator: string = Identities.Address.fromPublicKey(generatorPublicKey);
-
-            if (block.data.height === 1) {
-                delegate = new Wallet(generator);
-                delegate.publicKey = generatorPublicKey;
-
```

Comment:
It appears that this shortcut automatically creating wallet for the first block wasn't used